### PR TITLE
delete workflow

### LINKED ISSUE #532: Add DELETE workflow functionality
TBD if delete should be definitive or not.

### PR DESCRIPTION
**Pull request by DonHaul**

### CODE REVIEW

File: backoffice/backoffice/workflows/tests/cassettes/TestAirflowUtils.test_delete_workflow_dag_runs.yaml [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      - keep-alive
+      Content-Length:
+      - '0'
+    method: DELETE
+    uri: http://airflow-webserver:8080/api/v1/dags/author_create_initialization_dag/dagRuns/00000000-0000-0000-0000-000000000001
+  response:
+    body:
+      string: "{\n  \"detail\": \"DAGRun with DAG ID: 'author_create_initialization_dag'
+        and DagRun ID: '00000000-0000-0000-0000-000000000001' not found\",\n  \"status\":
+        404,\n  \"title\": \"Not Found\",\n  \"type\": \"https://airflow.apache.org/docs/apache-airflow/2.8.3/stable-rest-api-ref.html#section/Errors/NotFound\"\n}\n"
```
Comment (drjova) [inline backoffice/backoffice/workflows/tests/cassettes/TestAirflowUtils.test_delete_workflow_dag_runs.yaml:192]: there is 404 here, is this correct?

File: backoffice/backoffice/workflows/tests/cassettes/TestWorkflowViewSet.test_delete.yaml [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+      - application/problem+json
+      Date:
+      - Mon, 19 Aug 2024 13:16:38 GMT
+      Server:
+      - gunicorn
+      X-Robots-Tag:
+      - noindex, nofollow
+    status:
+      code: 404
+      message: NOT FOUND
```
Comment (drjova) [inline backoffice/backoffice/workflows/tests/cassettes/TestWorkflowViewSet.test_delete.yaml:144]: same

**Review by drjova — COMMENTED** [review]
Just few clarifications

File: backoffice/backoffice/workflows/tests/cassettes/TestWorkflowViewSet.test_delete.yaml (hunk 2 above)
Comment (DonHaul) [reply backoffice/backoffice/workflows/tests/cassettes/TestWorkflowViewSet.test_delete.yaml:144]: Yes. its due to the delete in the `teardown` trying to delete something that was already deleted during the test.

File: backoffice/backoffice/workflows/tests/cassettes/TestAirflowUtils.test_delete_workflow_dag_runs.yaml (hunk 1 above)
Comment (DonHaul) [reply backoffice/backoffice/workflows/tests/cassettes/TestAirflowUtils.test_delete_workflow_dag_runs.yaml:192]: dito

**Review by drjova — APPROVED** [review]

**Pull request merged**